File: app/src/main/java/no/nav/tilbakemeldingsmottak/interceptors/AbstractInterceptor.java
package no.nav.tilbakemeldingsmottak.interceptors;

import jakarta.servlet.http.HttpServletRequest;
import org.slf4j.MDC;
import org.springframework.web.servlet.HandlerInterceptor;

import static org.apache.commons.lang3.StringUtils.isBlank;

public class AbstractInterceptor implements HandlerInterceptor {

    public AbstractInterceptor() {
    }

    /**
     * @return Return the value in the first header that is defined in the request
     */
    String getHeaderValuesFromRequest(HttpServletRequest request, String fallbackValue, String... headerNames) {
        for (String headerName : headerNames) {
            String value = request.getHeader(headerName);
            if (!isBlank(value)) {
                return value;
            }
        }
        return fallbackValue;
    }

    void addValueToMDC(String value, String key) {
        if (value != null && !value.isEmpty()) {
            MDC.put(key, value);
        }
    }
}

File: app/src/main/kotlin/no/nav/tilbakemeldingsmottak/interceptors/AbstractInterceptor.kt
package no.nav.tilbakemeldingsmottak.interceptors

import jakarta.servlet.http.HttpServletRequest
import org.apache.commons.lang3.StringUtils
import org.slf4j.MDC
import org.springframework.web.servlet.HandlerInterceptor

open class AbstractInterceptor : HandlerInterceptor {
    /**
     * @return Return the value in the first header that is defined in the request
     */
    fun getHeaderValuesFromRequest(
        request: HttpServletRequest,
        fallbackValue: String,
        vararg headerNames: String?
    ): String {
        for (headerName in headerNames) {
            val value = request.getHeader(headerName)
            if (!StringUtils.isBlank(value)) {
                return value
            }
        }
        return fallbackValue
    }

    fun addValueToMDC(value: String?, key: String?) {
        if (!value.isNullOrEmpty()) {
            MDC.put(key, value)
        }
    }
}
